src/main/app/user_registration/ui.cljs

```diff
@@ -0,0 +1,81 @@
+  (dom/p :.text-danger message))
+
+(def ui-input-error (comp/computed-factory InputError {:keyfn :error/id}))
+
+(defsc UserRegistration
+  [this {:keys [:user-registration/id :user-registration/email
+                :user-registration/password :user-registration/confirm-password
+                :user-registration/errors] :as props}]
+  {:query         (fn [params]
+                    [:user-registration/id :user-registration/email
+                     :user-registration/password :user-registration/confirm-password
+                     {:user-registration/errors (comp/get-query InputError)}])
+   :ident         (fn []
+                    [:user-registration/id id])
+   :initial-state (fn [{:keys [id]}]
+                    {:user-registration/id (or id (random-uuid))
+                     :user-registration/errors []})}
+  (let [email-errors            (filter #(= :user-registration/email (:error/field-name %)) errors)
+        password-errors         (filter #(= :user-registration/password (:error/field-name %)) errors)
+        confirm-password-errors (filter #(= :user-registration/confirm-password (:error/field-name %)) errors)]
+    (dom/div
+      (dom/form {:onSubmit (fn [e] (.preventDefault e))}
+        (dom/div
+          (dom/input {:type     "text" :placeholder "Email"
+                      :onChange #(let [email (-> % .-target .-value)]
+                                   (comp/transact! this [(ur.mut/update-input
+                                                           {:user-registration/id    id
+                                                            :user-registration/email email})]))
+                      :onBlur   #(let [email (-> % .-target .-value)]
+                                   (comp/transact! this [(ur.mut/validate-input
+                                                          {:user-registration/id    id
+                                                           :user-registration/email email})]))})
+          (map #(ui-input-error %) email-errors))
+        (dom/div
+          (dom/input {:type     "password" :placeholder "Password"
+                      :onChange #(let [password (-> % .-target .-value)]
+                                   (comp/transact! this [(ur.mut/update-input
+                                                           {:user-registration/id       id
+                                                            :user-registration/password password})]))
+                      :onBlur   #(let [email (-> % .-target .-value)]
+                                   (comp/transact! this [(ur.mut/validate-input
+                                                           {:user-registration/id    id
+                                                            :user-registration/password email})]))
+                      })
+          (map #(ui-input-error %) password-errors))
+        (dom/div
+          (dom/input {:type     "password" :placeholder "Confirm password"
```

Comment:
Dispose of repetitions of `:onBlur` and `:onChange` functions in `input` elements.